extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/DataSourceReactiveMySQLConfig.java

```diff
@@ -38,4 +40,16 @@ public class DataSourceReactiveMySQLConfig {
      */
     @ConfigItem(defaultValueDocumentation = "disabled")
     public Optional<SslMode> sslMode = Optional.empty();
+
+    /**
+     * Connection timeout in seconds
+     */
+    @ConfigItem()
+    public OptionalInt connectionTimeout = OptionalInt.empty();
+
+    /**
+     * Select a different authentication plugin
+     */
+    @ConfigItem
+    public Optional<MySQLAuthenticationPlugin> authenticationPlugin = Optional.empty();
```

Comment:
I'm not sure how this is supposed to work. Did you test it locally?

Reply:
@gsmet 
Yes we tested it locally, passing the enum value as string i.e: quarkus.datasource.reactive.mysql.authentication-plugin=mysql_clear_password
will actually set the value to the to MySQLAuthenticationPlugin.MYSQL_CLEAR_PASSWORD, my guess is it's due to the MySQLAuthenticationPlugin enum constructor.

Reply:
@gsmet is there a place in the doc extension guide where it is explained how to handle config items in general and enums in particular?

Reply:
Sorry I missed it was a simple enum. I thought it would be a class to implement.